panel/models/video.ts

```diff
@@ -1,122 +1,133 @@
 import type * as p from "@bokehjs/core/properties"
+import type {StyleSheetLike} from "@bokehjs/core/dom"
+import {px} from "@bokehjs/core/dom"
 
 import {HTMLBox, HTMLBoxView} from "./layout"
+import video_css from "styles/models/video.css"
 
 export class VideoView extends HTMLBoxView {
   declare model: Video
 
-  protected videoEl: HTMLVideoElement
-  protected containerEl: HTMLElement
-  protected dialogEl: HTMLElement
-  private _blocked: boolean
-  private _time: any
-  private _setting: boolean
+  protected video_el: HTMLVideoElement
+
+  private _time: number
+  private _blocked: boolean = false
+  private _setting: boolean = false
 
   override initialize(): void {
     super.initialize()
-    this._blocked = false
-    this._setting = false
     this._time = Date.now()
   }
 
   override connect_signals(): void {
     super.connect_signals()
-    this.connect(this.model.properties.loop.change, () => this.set_loop())
-    this.connect(this.model.properties.paused.change, () => this.set_paused())
-    this.connect(this.model.properties.muted.change, () => this.set_muted())
-    this.connect(this.model.properties.autoplay.change, () => this.set_autoplay())
-    this.connect(this.model.properties.time.change, () => this.set_time())
-    this.connect(this.model.properties.value.change, () => this.set_value())
-    this.connect(this.model.properties.volume.change, () => this.set_volume())
+
+    const {loop, paused, muted, autoplay, time, value, volume} = this.model.properties
+    this.on_change(loop, () => this.set_loop())
+    this.on_change(paused, () => this.set_paused())
+    this.on_change(muted, () => this.set_muted())
+    this.on_change(autoplay, () => this.set_autoplay())
+    this.on_change(time, () => this.set_time())
+    this.on_change(value, () => this.set_value())
+    this.on_change(volume, () => this.set_volume())
+  }
+
+  override stylesheets(): StyleSheetLike[] {
+    return [...super.stylesheets(), video_css]
   }
 
   override render(): void {
     super.render()
-    this.videoEl = document.createElement("video")
-    this.containerEl = document.createElement("div")
-    this.containerEl.className="pn-video-container"
-    this.containerEl.style.height = "100%"
-    this.containerEl.style.width = "100%"
-    this.videoEl.style.objectFit = "fill"
-    this.videoEl.style.width = "100%"
-    this.videoEl.style.height = "100%"
-    if (!this.model.sizing_mode || this.model.sizing_mode === "fixed") {
-      if (this.model.height) {
-        this.videoEl.height = this.model.height
+
+    this.video_el = document.createElement("video")
+    const container_el = document.createElement("div")
+    container_el.className = "pn-video-container"
+    container_el.style.height = "100%"
+    container_el.style.width = "100%"
```

Comment:
I feel like this wrapper element is unnecessary, but maybe there's a reason for this. If there's one, it would be good to explain it here.